evm/evm.go

```diff
@@ -299,31 +302,22 @@ func defaultChainConfig(enableConstantinople bool) params.ChainConfig {
 	}
 }
 
-func defaultVmConfig(evmDebuggingEnabled bool) vm.Config {
-	logCfg := vm.LogConfig{
-		DisableMemory:  true, // disable memory capture
-		DisableStack:   true, // disable stack capture
-		DisableStorage: true, // disable storage capture
-		Limit:          0,    // maximum length of output, but zero means unlimited
-	}
-	debug := false
-
-	if evmDebuggingEnabled {
-		log.Error("WARNING!!!! EVM Debug mode enabled, do NOT run this on a production server!!!")
-		logCfg = vm.LogConfig{
+func createVmConfig(tracer vm.Tracer) (vm.Config, error) {
+	if tracer == nil {
+		logCfg := vm.LogConfig{
 			DisableMemory:  true, // disable memory capture
 			DisableStack:   true, // disable stack capture
 			DisableStorage: true, // disable storage capture
 			Limit:          0,    // maximum length of output, but zero means unlimited
 		}
-		debug = true
+		tracer = vm.NewStructLogger(&logCfg)
 	}
-	logger := vm.NewStructLogger(&logCfg)
+
 	return vm.Config{
 		// Debug enabled debugging Interpreter options
-		Debug: debug,
+		Debug: tracer != nil,
```

Comment:
`tracer` will never be `nil` here because when `createVmConfig(nil)` is called a new tracer is created using `vm.NewStructLogger(&logCfg)`.